### PR TITLE
8240343: JDI stopListening/stoplis001 "FAILED: listening is successfully stopped without starting listening"

### PR DESCRIPTION
**Pull request by alexmenkov**
The test starts listening on dynamic port and calls stopListening with incorrect (cArgs1) and correct (cArgs2) argument maps.
Incorrect map is created by finding "free" port (`(new ServerSocket(0)).getLocalPort()`)
The test fails if the same port is selected later when the listening starts.
To avoid this free port should be picked after listening on dynamic port started.

Additionally removed unnecessary `exclusiveAccess.dirs=.` test property and incorrect comment in the description.
The test uses `SocketListen` connectors, the coment is about `SharedMemoryListen` connector.

Testing: run the test on all Oracle-supported platforms 100 times

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8240343](https://bugs.openjdk.org/browse/JDK-8240343): JDI stopListening/stoplis001 "FAILED: listening is successfully stopped without starting listening" (**Bug** - P4)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [3b0d5540](https://git.openjdk.org/jdk/pull/18705/files/3b0d55404ca67869e75d98b68c3829d5a25d2ee5)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18705/head:pull/18705` \
`$ git checkout pull/18705`

Update a local copy of the PR: \
`$ git checkout pull/18705` \
`$ git pull https://git.openjdk.org/jdk.git pull/18705/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18705`

View PR using the GUI difftool: \
`$ git pr show -t 18705`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18705.diff">https://git.openjdk.org/jdk/pull/18705.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18705#issuecomment-2046246690)

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/ListeningConnector/stopListening/stoplis001.java [hunk 1]
```diff
@@ -96,6 +95,9 @@ private int runIt(String argv[], PrintStream out) {
         else
             log.display("TEST: start listening the address " + addr);
 
+        // argHandler.getTransportPort() returns a free port (different from the listening port in cArgs2)
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/ListeningConnector/stopListening/stoplis001.java:98]:
```suggestion
        // argHandler.getTransportPort() returns a free port (different from the port allocated by startListen(cArgs2))
```

Comment (alexmenkov) [reply test/hotspot/jtreg/vmTestbase/nsk/jdi/ListeningConnector/stopListening/stoplis001.java]: done

**Review by plummercj — APPROVED** [review]

**Review by sspitsyn — APPROVED** [review]
Looks good to me.

**Comment by plummercj** [comment]
You might want to consider adding the following diff. It will help in case this issue (or similar) reproduces again:

```
--- a/test/hotspot/jtreg/vmTestbase/nsk/jdi/ListeningConnector/stopListening/stoplis001.java
+++ b/test/hotspot/jtreg/vmTestbase/nsk/jdi/ListeningConnector/stopListening/stoplis001.java
@@ -92,6 +92,8 @@ private int runIt(String argv[], PrintStream out) {
         }
         else
             log.display("TEST: start listening the address " + addr);
+        log.display("cArgs1: " + cArgs1);
+        log.display("cArgs2: " + cArgs2);
 
 /* Check that an Exception is thrown if ListeningConnector.stopListening
  has been invoked with argument map different from the one given for
```

```
 TEST: start listening the address localhost:42473
cArgs1: {timeout=timeout=, port=port=45521, localAddress=localAddress=}
cArgs2: {timeout=timeout=, port=port=42473, localAddress=localAddress=} 
```

**Comment by alexmenkov** [comment]
> You might want to consider adding the following diff. It will help in case this issue (or similar) reproduces again:
> 
makes sense. added.

**Review by plummercj — APPROVED** [review]

**Pull request closed without merging**